---
title: Устранение проблем безопасности и контроля доступа
description: Узнайте, как устранять проблемы безопасности и контроля доступа в фабрике данных Azure.
services: data-factory
author: lrtoyou1223
ms.service: data-factory
ms.topic: troubleshooting
ms.date: 11/19/2020
ms.author: lle
ms.reviewer: craigg
ms.openlocfilehash: 99c03ae4430d1a4caf575bdb9900200af0217bf1
ms.sourcegitcommit: 6172a6ae13d7062a0a5e00ff411fd363b5c38597
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 12/11/2020
ms.locfileid: "97109777"
---
# <a name="troubleshoot-azure-data-factory-security-and-access-control-issues"></a>Устранение неполадок, связанных с управлением безопасностью и доступом в фабрике данных Azure

[!INCLUDE[appliesto-adf-asa-md](includes/appliesto-adf-asa-md.md)]

В этой статье рассматриваются распространенные способы устранения неполадок безопасности и контроля доступа в фабрике данных Azure.

## <a name="common-errors-and-messages"></a>Распространенные ошибки и сообщения


### <a name="connectivity-issue-of-copy-activity-in-cloud-data-store"></a>Проблемы с подключением к действию копирования в облачном хранилище данных

#### <a name="symptoms"></a>Симптомы

При возникновении проблем с подключением к источнику или приемнику данных могут возвращаться различные типы сообщений об ошибках.

#### <a name="cause"></a>Причина 

Проблема в основном обусловлена следующими факторами:

1. Параметр прокси-сервера в узле ИК-связи с локальным размещением (если используется локальная среда IR)

1. Параметр брандмауэра в узле ИК-связи с локальным размещением (если используется локальная среда IR)

1. Параметр брандмауэра в облачном хранилище данных

#### <a name="resolution"></a>Решение

1. Проверьте следующие моменты, чтобы убедиться, что неполадка вызвана проблемой подключения:

   - Ошибка возникает из соединителей источника или приемника.

   - Действие завершается с ошибкой в начале копии

   - Это последовательный сбой для Azure IR или самостоятельно размещенного IR с одним узлом, так как он может быть случайным сбоем для автономно размещенных IR с несколькими узлами, если проблема возникает только в части узлов.

1. Проверьте параметры прокси-сервера, брандмауэра и сети, если используется локальная **Среда IR** , так как запуск в одном хранилище данных может быть выполнен в Azure IR. Для устранения неполадок обратитесь к следующим ссылкам:

   Локальные [ИК-порты и брандмауэры](https://docs.microsoft.com/azure/data-factory/create-self-hosted-integration-runtime#ports-and-firewalls) 
    [Соединитель ADLS](https://docs.microsoft.com/azure/data-factory/connector-azure-data-lake-store)
  
1. Если вы используете **Azure IR**, попробуйте отключить параметр брандмауэра хранилища данных. Таким образом можно устранить проблемы следующих двух обстоятельств:
  
   * [Azure IR IP-адреса](https://docs.microsoft.com/azure/data-factory/azure-integration-runtime-ip-addresses) отсутствуют в списке разрешений.

   * *Разрешить доверенным службам Майкрософт доступ к этой учетной записи хранения* отключена для [хранилища BLOB-объектов Azure](https://docs.microsoft.com/azure/data-factory/connector-azure-blob-storage#supported-capabilities) и [ADLS поколения 2](https://docs.microsoft.com/azure/data-factory/connector-azure-data-lake-storage#supported-capabilities).

   * *Разрешение доступа к службам Azure* не включено для ADLS 1-го поколения.

1. Если приведенные выше методы не работают, обратитесь за помощью в корпорацию Майкрософт.


### <a name="invalid-or-empty-authentication-key-issue-after-public-network-access-is-disabled"></a>Недействительная или пустая ошибка ключа проверки подлинности после отключения доступа к общедоступной сети

#### <a name="symptoms"></a>Симптомы

После отключения доступа к общедоступной сети для фабрики данных, так как локальная среда выполнения интеграции вызывает следующую ошибку: "ключ проверки подлинности недопустим или пуст".

#### <a name="cause"></a>Причина

Проблема, скорее всего, вызвана проблемой с разрешением DNS, так как отключение общедоступного подключения и установка частной конечной точки не помогают повторно подключиться.

Чтобы проверить, разрешено ли полное доменное имя фабрики данных в общедоступный IP-адрес, выполните следующие действия.

1. Убедитесь, что виртуальная машина Azure создана в той же виртуальной сети, что и частная конечная точка фабрики данных.

2. Выполните PsPing и выполните проверку связи из виртуальной машины Azure с полным доменным именем фабрики данных:

   `ping <dataFactoryName>.<region>.datafactory.azure.net`

   `psping.exe <dataFactoryName>.<region>.datafactory.azure.net:443`

   > [!Note]
   > Необходимо указать порт для команды PsPing, тогда как порт 443 не является обязательным.

3. Проверьте, разрешены ли обе команды на общедоступный IP-адрес ADF, основанный на указанном регионе (формат XXX. xxx. xxx. 0).

#### <a name="resolution"></a>Решение

- Вы можете обратиться к статье [о службе "Частная связь Azure" с фабрикой данных Azure](https://docs.microsoft.com/azure/data-factory/data-factory-private-link#dns-changes-for-private-endpoints). Инструкция предназначена для настройки частного DNS-зоны или сервера для разрешения полного доменного имени фабрики данных в частный IP-адрес.

- Если вы не хотите настраивать частную зону или сервер DNS в настоящее время, выполните указанные ниже действия как временное решение. Однако пользовательская служба DNS по-прежнему рекомендуется в долгосрочном решении.

  1. Измените файл узла в Windows и сопоставьте частный IP-адрес (закрытая конечная точка ADF) с полным доменным именем ADF.
  
     Перейдите по пути "C:\Windows\System32\drivers\etc" на виртуальной машине Azure и откройте файл **узла** в блокноте. Добавьте строку сопоставления частный IP-адрес в полное доменное имя в конце файла и сохраните изменения.
     
     ![Добавление сопоставления с узлом](media/self-hosted-integration-runtime-troubleshoot-guide/add-mapping-to-host.png)

  1. Повторно выполните те же команды в приведенных выше шагах проверки, чтобы проверить ответ, который должен содержать частный IP-адрес.

  1. Повторно зарегистрируйте локальную среду выполнения интеграции, и проблема должна быть решена.
 

### <a name="unable-to-register-ir-authentication-key-on-self-hosted-vms-due-to-private-link"></a>Не удалось зарегистрировать ключ проверки подлинности IR на самостоятельно размещенных виртуальных машинах из-за частной ссылки

#### <a name="symptoms"></a>Симптомы

Не удалось зарегистрировать ключ проверки подлинности IR на виртуальной машине, размещенной на собственном сервере, так как включена частная ссылка.

Сведения об ошибке показаны ниже:

`
Failed to get service token from ADF service with key *************** and time cost is: 0.1250079 seconds, the error code is: InvalidGatewayKey, activityId is: XXXXXXX and detailed error message is Client IP address is not valid private ip Cause Data factory couldn’t access the public network thereby not able to reach out to the cloud to make the successful connection.
`

#### <a name="cause"></a>Причина

Эта ошибка может быть вызвана виртуальной машиной, в которой вы пытаетесь установить локальную среду IR. Чтобы подключиться к облаку, убедитесь, что включен доступ к общедоступной сети.

#### <a name="resolution"></a>Решение

 **Решение 1.** Чтобы устранить проблему, выполните указанные ниже действия.

1. Перейдите на страницу [фабрики — обновление](https://docs.microsoft.com/rest/api/datafactory/Factories/Update) .

1. В правом верхнем углу нажмите кнопку **попробовать** .

1. В разделе **Параметры** заполните необходимые сведения. 

1. В разделе **текст** вставьте следующее свойство:
    ```
    { "tags": { "publicNetworkAccess":"Enabled" } }
    ``` 

1. Выберите **выполнить** , чтобы запустить функцию. 

1. Убедитесь, что отображается **код отклика: 200** . Вставленное свойство должно также отображаться в определении JSON.

1. Добавьте ключ проверки подлинности IR еще раз в среду выполнения интеграции.


**Решение 2.** Для решения можно обратиться к следующей статье:

https://docs.microsoft.com/azure/data-factory/data-factory-private-link

Попробуйте включить доступ к общедоступной сети в пользовательском интерфейсе, как показано на следующем снимке экрана:

![Включить доступ к общедоступной сети](media/self-hosted-integration-runtime-troubleshoot-guide/enable-public-network-access.png)

## <a name="next-steps"></a>Дальнейшие действия

Для получения дополнительных сведений об устранении неполадок воспользуйтесь следующими ресурсами:

*  [Частная ссылка для фабрики данных](data-factory-private-link.md)
*  [Блог о Фабрике данных](https://azure.microsoft.com/blog/tag/azure-data-factory/)
*  [Запросы на добавление функции в Фабрику данных](https://feedback.azure.com/forums/270578-data-factory)
*  [Видео по Azure](https://azure.microsoft.com/resources/videos/index/?sort=newest&services=data-factory)
*  [Microsoft Q&странице](/answers/topics/azure-data-factory.html)
*  [Форум о переполнении стека для фабрики данных](https://stackoverflow.com/questions/tagged/azure-data-factory)
*  [Сведения о Фабрике данных в Twitter](https://twitter.com/hashtag/DataFactory)
